src/contract_wrappers/modules/checkpoint/voting/advanced_p_l_c_r_voting_checkpoint_wrapper/common.ts

```diff
@@ -0,0 +1,1206 @@
+  >;
+  (params: GetVotersExemptedLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointVotersExemptedEventArgs_3_1_0>[]
+  >;
+  (params: GetVoteCommitLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointVoteCommitEventArgs_3_1_0>[]
+  >;
+  (params: GetVoteRevealedLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointVoteRevealedEventArgs_3_1_0>[]
+  >;
+  (params: GetBallotCancelledLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointBallotCancelledEventArgs_3_1_0>[]
+  >;
+  (params: GetChangedBallotExemptedVotersListLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointChangedBallotExemptedVotersListEventArgs_3_1_0>[]
+  >;
+  (params: GetChangedDefaultExemptedVotersListLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointChangedDefaultExemptedVotersListEventArgs_3_1_0>[]
+  >;
+  (params: GetPauseLogsAsyncParams): Promise<LogWithDecodedArgs<AdvancedPLCRVotingCheckpointPauseEventArgs_3_1_0>[]>;
+  (params: GetUnpauseLogsAsyncParams): Promise<
+    LogWithDecodedArgs<AdvancedPLCRVotingCheckpointUnpauseEventArgs_3_1_0>[]
+  >;
+}
+
+/**
+ * @param name Name of the ballot (Should be unique)
+ * @param startTime startTime of the ballot
+ * @param commitDuration Unix time period till the voters commit there vote
+ * @param revealDuration Unix time period till the voters reveal there vote starts when commit duration ends
+ */
+export interface BallotParams extends TxParams {
+  name: string;
+  startTime: Date;
+  commitDuration: number;
+  revealDuration: number;
+}
+
+/**
+ * @param proposalTitles Title of proposal
+ * @param proposalDetails Off-chain details related to the proposal
+ * @param choices Choices of proposals
+ * @param choicesCounts No. of choices (If it is 0 then it means NAY/YAY ballot type is choosen).
+ */
+export interface CreateStatutoryBallotParams extends BallotParams {
+  proposalTitles: string[];
+  proposalDetails: string;
+  choices: string[];
+  choicesCounts: number;
+}
+
+/**
+ * @param checkpointId Valid checkpoint Id
+ */
+export interface CreateCustomStatutoryBallotParams extends CreateStatutoryBallotParams {
+  checkpointId: number;
+}
+
+/**
+ * @param proposalTitles Title of proposal
+ * @param proposalDetails Off-chain details related to the proposal
+ * @param choices Choices of proposals
+ * @param choicesCounts Array of No. of choices (If it is 0 then it means NAY/YAY ballot type is choosen).
+ */
+export interface CreateCumulativeBallotParams extends BallotParams {
+  proposalTitles: string[];
+  proposalDetails: string[];
+  choices: string[];
+  choicesCounts: number[];
+}
+
+/**
+ * @param choicesCounts No. of choices (If it is 0 then it means NAY/YAY ballot type is choosen).
+ * @param checkpointId Valid checkpoint Id
+ */
+export interface CreateCustomCumulativeBallotParams extends CreateCumulativeBallotParams {
+  choicesCounts: number[];
```

Comment:
Im not sure about this one, isn't this `choicesCounts` variable already included in `CreatCumulativeBallotParams`

Reply:
good catch